### PR TITLE
dotCMS/core#20182 Create the navigation menu

### PR DESCRIPTION
**Pull request by rjvelazco**

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -4,36 +4,34 @@ import Link from 'next/link';
...
-    const [res] = useQuery({ query: NAVIGATION_MENU_QUERY });
-    const { data } = res;
+export default function Home({ data }: DotcmsDocumentationCollection): JSX.Element {
     return (
         <Container>
             <Head>
                 <title>Documentation</title>
                 <link href="/favicon.ico" rel="icon" />
             </Head>
+            <h1>Hey</h1>
```
Comment (fmontes) [inline pages/index.tsx]: remove

File: pages/index.tsx [hunk 2]
```diff
@@ -45,31 +43,36 @@ const DotCollection = ({ data }: any) => {
...
-
+export async function getStaticProps(): Promise<NavigationProp> {
+    const client = new GraphQLClient(BASE_URL);
+    const data = await client.request(NAVIGATION_MENU_QUERY);
     return {
         props: {
-            urqlState: ssrCache.extractData()
-        },
-        revalidate: 600
+            data
```
Comment (fmontes) [inline pages/index.tsx]:
```suggestion
            data: data. DotcmsDocumentationCollection
```

File: pages/index.tsx [hunk 3]
```diff
@@ -4,36 +4,34 @@ import Link from 'next/link';
...
+            <h1>Hey</h1>
             <nav>
                 <DotCollection data={data.DotcmsDocumentationCollection[0]} />
             </nav>
         </Container>
     );
 }
 
-const DotCollection = ({ data }: any) => {
+const DotCollection = ({ data }: DotcmsDocumentationData) => {
```
Comment (fmontes) [inline pages/index.tsx]:
```suggestion
const DotCollection = ({ data }: DotcmsDocumentation[]) => {
```

File: pages/index.tsx [hunk 4]
```diff
@@ -4,36 +4,34 @@ import Link from 'next/link';
...
     if (!data.dotcmsdocumentationchildren) {
         return null;
     }
 
     return (
         <ul>
-            {data.dotcmsdocumentationchildren.map((item) => (
+            {data.dotcmsdocumentationchildren.map((item: DotcmsDocumentation) => (
```
Comment (fmontes) [inline pages/index.tsx:33]:
```suggestion
            {data.map((item: DotcmsDocumentation) => (
```

File: pages/index.tsx [hunk 5]
```diff
@@ -45,31 +43,36 @@ const DotCollection = ({ data }: any) => {
...
+
+interface DotcmsDocumentationCollection {
+    data: {
+        DotcmsDocumentationCollection: DotcmsDocumentation[];
+    };
+}
+
+interface DotcmsDocumentationData {
+    data: DotcmsDocumentation;
+}
```
Comment (fmontes) [inline pages/index.tsx]:
```suggestion
```

**Review by fmontes — CHANGES_REQUESTED** [review]

File: pages/index.tsx (hunk 4 above)
Comment (rjvelazco) [reply pages/index.tsx:33]: Why did you change 'data.dotcmsdocumentationchildren' to 'data'? data.dotcmsdocumetnationchildren is the array we are iterating.

**Pull request merged**